_layouts/home.html

```diff
@@ -2,6 +2,23 @@
 layout: archive
 ---
 
+{% if page.url == "/" %}
+  {% assign entries_layout = page.entries_layout | default: 'list' %}
+  {% assign print_header = true %}
+  <div class="entries-{{ entries_layout }}">
+    {% assign posts = site.posts %}
+    {% for post in posts %}
+      {% if post.pinned %}
+        {% if print_header %}
```

Comment:
This is also unnecessary and can be done in two other ways:

- If the page-level switch is enabled, then a header can be displayed even with no posts (users should handle this on their own).
- If the filtered result (where expression) is non-empty, then show the header.